changelog/5986.improvement.md

```diff
@@ -0,0 +1,3 @@
+Using the `TrainingDataImporter` interface to load the data in `rasa test core`.
+
+Referring to failed test stories via absolute path (change in test `test_source_in_failed_stories`).
```

Comment:
There is no need to point to the actual code change you did. If users want to see that than they can find the PR using the issue number of the changelog. Rasa developers can also find the PR by its commit hash.

```suggestion
Failed test stories are now referenced by their absolute path instead of the relative path.
```